### PR TITLE
add formatter docs; minor adjustments for formatter syntax

### PR DESCRIPTION
**Pull request by derekmorr**
This adds docs for formatters. I made some changes since the last PR to make the syntax more consistent.

I'd appreciate feedback on the docs. It can get complicated with unitgroups, formatters, implicits, etc. I'm not sure I structured it in the best way.

### CODE REVIEW

File: shared/src/main/tut/README.md [hunk 1]
```diff
@@ -709,6 +709,123 @@ The `UnitGroup` values provided with Squants are only samples and aren't intende
 We encourage users to make their own `UnitGroup` defintitions and submit them as PRs if they're generally
 applicable.
 
+## Formatters
+
+Squants provides an experimental API for formatting Quantities in the "best unit." For example,
```
Comment (cquiroz) [inline shared/src/main/tut/README.md]: I wonder a bit about the `experimental` part. If this is something we may change then the `experimental` tag fits very well but then perhaps it should be on an `experimental` package.

Do you foresee possible changes in the near future that could break the API?

File: shared/src/main/tut/README.md [hunk 2]
```diff
@@ -709,6 +709,123 @@ The `UnitGroup` values provided with Squants are only samples and aren't intende
...
+To use `DefaultFormatter` import it, and a unit group:
+
+```tut:silent
+import squants.formatter.DefaultFormatter
+import squants.unitgroups.misc.AstronomicalLengthUnitGroup
+```
+
+Then create the formatter by passing in a unit group:
+```tut:book
+val astroFormatter = new DefaultFormatter[Length] { val unitGroup = AstronomicalLengthUnitGroup }
```
Comment (cquiroz) [inline shared/src/main/tut/README.md]: Why this isn't built just as a regular class? `new DefaultFormatter[Length](AstronomicalLengthUnitGroup)`

File: shared/src/main/tut/README.md (hunk 1 above)
Comment (derekmorr) [inline shared/src/main/tut/README.md]: I'm not sure. We've not had any feedback on the API from people who requested it. Once it's released, it's possible we'll want to change it based on feedback.

I'm fine moving unitgroups and formatters into an experimental package.

File: shared/src/main/tut/README.md (hunk 2 above)
Comment (derekmorr) [inline shared/src/main/tut/README.md]: Probably b/c everything else was a trait. I agree the class syntax is better.

File: shared/src/main/tut/README.md (hunk 1 above)
Comment (cquiroz) [inline shared/src/main/tut/README.md]: I guess either way the API will change but the experimental would at least be an indication.
I don't have a strong opinion either way

Comment (derekmorr) [inline shared/src/main/tut/README.md]: I'll refactor it into an experimental package and update the README.

**Comment by derekmorr** [comment]
I pushed up a commit that addresses the latest comments.

**Comment by cquiroz** [comment]
that seems good, we can move it to non `experimental` on the next release

**Comment by derekmorr** [comment]
Are we ok to merge this then?

**Comment by cquiroz** [comment]
I think so

**Pull request merged**